src/main/java/bio/terra/service/duos/DuosService.java

```diff
@@ -196,8 +186,8 @@ public DuosFirecloudGroupsSyncResponse syncDuosDatasetsAuthorizedUsers() {
    * @param errors collection to be supplemented with any error encountered during sync
    * @return id of the successfully synced Firecloud group's record
    */
-  private UUID syncFirecloudGroupContents(
-      DuosFirecloudGroupModel firecloudGroup, List<ErrorModel> errors) {
+  @VisibleForTesting
+  UUID syncFirecloudGroupContents(DuosFirecloudGroupModel firecloudGroup, List<ErrorModel> errors) {
```

Comment:
Looking at this again, this method actually returns either a `UUID` or an `ErrorModel`. In Scala there's a class called `Either` that represents this concept, but I don't think java has an analogous object.

For this use case, you could return a local `record` that contains both `UUID` and `ErrorModel` and you could `map` & `filter` over it to get the IDs and the errors. Doing that would avoid the need for a synchronized `ArrayList` for `errors`, and also avoid the need to write a method that modifies its arguments.

Reply:
Interesting idea, I'll take a look.